File: _java_/src/main/java/patterns/builder/Car.java
package patterns.builder;

public class Car {
    private String maker;
    private Transmission transmission;
    private int maxSpeed;

    public void setMaker(String maker) {
        this.maker = maker;
    }

    public void setTransmission(Transmission transmission) {
        this.transmission = transmission;
    }

    public void setMaxSpeed(int maxSpeed) {
        this.maxSpeed = maxSpeed;
    }

    @Override
    public String toString() {
        return "Car [maker=" + maker + ", transmission=" + transmission
                + ", maxSpeed=" + maxSpeed + "]";
    }
}

File: _kotlin_/src/main/kotlin/patterns/builder/Car.kt
package patterns.builder

class Car {

    var maker: String? = null
    var transmission: Transmission? = null
    var maxSpeed = 0

    override fun toString(): String {
        return ("Car [make=" + maker + ", transmission=" + transmission
            + ", maxSpeed=" + maxSpeed + "]")
    }
}

enum class Transmission {
    MANUAL, AUTO
}
